ksqldb-engine/src/main/java/io/confluent/ksql/services/KafkaConsumerGroupClientImpl.java

```diff
@@ -117,10 +115,11 @@ public void deleteConsumerGroups(final Set<String> groups) {
       ExecutorUtil.executeWithRetries(
           () -> adminClient.get().deleteConsumerGroups(groups).all().get(),
           e -> (e instanceof RetriableException)
-              || (e instanceof GroupNotEmptyException && retryCount.getAndIncrement() < 5),
-          () -> Duration.of(3, ChronoUnit.SECONDS)
+              || (e instanceof GroupNotEmptyException),
+          (retry) -> Duration.of(3L * retry, ChronoUnit.SECONDS),
```

Comment:
First change: add linear backoff so that we will wait 3, 6, 9, and so forth seconds between each retry instead of just waiting 3 seconds every time.

Reply:
Just for my own curiosity, how did we come up with using 3 seconds?

Reply:
There's a comment there that says that it's the default heartbeat interval. I'm not sure if the person who made this choice was aware that we're actually waiting for the heartbeat interval + the session timeout, though.

Come to think of it, I bet that this thing has just recently started timing out as another casualty of the KIP to change the default session interval to 30 seconds.

Reply:
Ah i only saw the first half of the comment so I missed the explanation 😅  And that makes sense, good catch!

Reply:
> I'm not sure if the person who made this choice was aware that we're actually waiting for the heartbeat interval + the session timeout, though.

hehe, guilty

Reply:
Aha! Smoked you out!

Anyway, I think it was fine until we bumped that default session interval up so high.